### PR TITLE
Ensure mockbuild UID and homedir permissions are correct after mock init from root_cache

### PR DESCRIPTION
**Pull request by adamel**
Fixes issue #48.

### CODE REVIEW

**Comment by xsuchy** [comment]
The idea is good. The implementation is sub-optimal. You should use
```
  # maybe elevate privs here using self.buildroot.uid_manager and then
  self.buildroot.uid_manager.changeOwner(self.homedir, recursive=True)
```

**Comment by adamel** [comment]
I can do that, but it will change the existing code in chown_home_dir to follow symlinks (uid_manager.changeOwner() uses os.chown() instead of of.lchown()) and symlinks might point to something outside the mock root.

Is there a reason for uid_manager._tolerant_chown() not using os.lchown() ?

**Comment by xsuchy** [comment]
I am not aware of any such reason.

**Comment by adamel** [comment]
Pushed updated branch. Also optimized it so we do not do a full recursive traverse of the entire homedir if it is already owned by the correct user.

File: py/mockbuild/buildroot.py [hunk 1]
```diff
@@ -161,6 +161,13 @@ def _init(self, prebuild, do_log):
                 self.state.start(update_state)
                 self.pkg_manager.update()
                 self.state.finish(update_state)
+        else:
+            self._fixup_build_user()
```
Comment (xsuchy) [inline py/mockbuild/buildroot.py]: Instead of this call self.uid_manager.dropPrivsTemp() and at the enf of else block self.uid_manager.restorePrivs()

Comment (adamel) [inline py/mockbuild/buildroot.py]: What will that achieve? The purpose of _fixup_build_user() is (only) to modify the /etc/passwd file in the mock root, and as far as I can see dropPrivsTemp() does nothing like that?

**Comment by xsuchy** [comment]
Ah, now I see what you are trying to achieve. Merged. Thank you.

**Pull request closed without merging**